src/test/java/com/midaswebserver/midasweb/services/LoginServiceImpTest.java

```diff
@@ -0,0 +1,95 @@
+package com.midaswebserver.midasweb.services;
+
+import com.midaswebserver.midasweb.forms.LoginForm;
+import com.midaswebserver.midasweb.models.User.User;
+import com.midaswebserver.midasweb.repositories.UserRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.util.List;
+
+import static org.springframework.test.util.AssertionErrors.*;
+
+@SpringBootTest
+public class LoginServiceImpTest {
+    //get services and other required objects to test or to aid the testing
+    @Autowired
+    private LoginService loginService;
+    @Autowired
+    private UserRepository userRepo;
+
+    @BeforeEach
+    public void beforeTest() {
+        String username = "testuser";
+        String password = "testpass";
+        User fakeUser = new User(username, Integer.toString(password.hashCode()));//login (the name) might want to be changed as it is a little confusing
+        assertNotNull("UserRepository must be injected", userRepo);
+        assertNotNull("loginService must be injected", loginService);
+
+        // Ensure dummy record is in the DB
+        final List<User> users =
+                userRepo.findByUsernameIgnoreCase(username);
+        if (users.isEmpty())
+            userRepo.save(fakeUser);
+    }
+
+    @Test//happy
+    public void validateUserSuccessTest1() {
+        String username = "testuser";
```

Comment:
If you're going to insert a record in @BeforeTest, then you should make the username global to the entire test class to make sure that each test uses the same String, rather than expecting each test user to declare it locally (and hopefully not accidentally use a different value).